sockeye/training.py

```diff
@@ -761,6 +767,9 @@ def _cleanup(self, lr_decay_opt_states_reset: str, process_manager: Optional['De
                 self.tflogger.log_metrics(decoder_metrics, decoded_checkpoint)
             utils.write_metrics_file(self.state.metrics, self.metrics_fname)
 
+        if keep_training_state:
```

Comment:
matter of taste, but I would prefer to move the following lines under an `if not keep_training_state` instead of the early return.

Reply:
fixed